packages/lib/file-api-driver-amazon-s3.js

```diff
@@ -219,8 +219,8 @@ class FileApiDriverAmazonS3 {
 				const filePath = options.path;
 				if (!filePath) throw new Error('get: target options.path is missing');
 
-				// TODO: check if this ever hits on RN
-				await shim.fsDriver().writeBinaryFile(filePath, output);
+				const buffer = Buffer.from(output);
+				await shim.fsDriver().writeFile(filePath, buffer.toString('base64'));
```

Comment:
Wouldn't this cause issues for large attachments?
Unless I'm reading it wrong, this will make Joplin to load 2 copies of the file in memory one of which is also base64 encoded.

Reply:
Yes, this isn't the right thing to do for large attachments. It is enough to attach pictures and such to notes.

Support for large attachments requires to rewrite the S3 synchronization to make use of the streams which probably also introduces another usage of RNFetchBlob for the mobile app. If that is acceptable for everyone I can do a follow-up PR to optimize this.

Reply:
Also the call to writeBinaryFile here is the right one, and how each app write this data to disk is implementation detail that should be done in the driver file. So in that case you should leave this code as it is, and modify `writeBinaryFile` in fs-driver-rn to make it do the right thing. As Roman mentioned, it would be best if you could write the data directly without converting it first to Base64. I believe rn-fetch-blob can do that.

Reply:
To my knowledge there is no support in RNFetchBlob or RNFS to write binary data. I've implemented `writeBinaryFile` now with the help of RNFetchBlob's FileStream API and implemented a stream Writable doing the base64 encoding. This addresses @roman-r-m 2 copies issue of having 2 copies of the file in memory.